src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,183 @@
+import { chakra, useStyleConfig } from "@chakra-ui/react";
+import List from "../List/List";
+import Link from "../Link/Link";
+import Icon, { IconSizes } from "../Icons/Icon";
+import { LayoutTypes } from "../../helpers/types";
+import { socialMediaDataMap } from "./SocialMediaLinksUtils";
+import React, { forwardRef } from "react";
+
+export const borderTypeArray = ["none", "circular", "straight"] as const;
+export type BorderType = typeof borderTypeArray[number];
+
+export const colorTypeArray = ["link", "textDefault", "textInverse"] as const;
+export type ColorType = typeof colorTypeArray[number];
+
+export const sizeTypeArray = ["small", "medium", "large"] as const;
+export type SizeType = typeof sizeTypeArray[number];
+
+// This list of allowed types is defined in the SocialMediaDataMap found in SocialMediaLinksUtils.ts and extracted here.
+export const socialMediaLinkTypeArray = socialMediaDataMap.map(
+  ({ type }) => type
+);
+export type SocialMediaLinkType = typeof socialMediaLinkTypeArray[number];
+
+// @todo If SocialMediaLinkDataType equals this, and we only actually use
```

Comment:
Hmm we missed this before then, but you're right we don't need both. I prefer the name `SocialMediaLinkDataProps` but either can be used.

Reply:
This refactoring was a result of [this conversation](https://github.com/NYPL/nypl-design-system/pull/1391#discussion_r1308028537). I am not yet well versed enough in TS to completely understand if we should choose `interface` or `type` or have both as different constants. Please advise.

Reply:
For this specific case, they're interchangeable. So you could say
```
linksData?: SocialMediaLinkDataProps[];
```
instead of 
```
linksData?: SocialMediaLinkDataType[];
```

That would eliminate the use of `SocialMediaLinkDataType`. The type is more for primitive variables and values and the interface is more for a contractual structure definition. Interfaces are easier to extend so we like to use them for component props. So if we have a base component with a certain structure of props, and then we have a similar component that needs more props, we can extend the base interface and reuse it. You could do the same with types but it gets trickier. For types you can do the intersection `&` to merge types into a new one. 

So my _preference_ would be to stick with the interface. If a consuming app needs to extend the base prop interface, they can. As an example, if DXP then needs its own very specific social media component, you could import the `SocialMediaLinkDataProps` interface and extend it with your own props. Maybe DXP now needs a `donateUrl` prop for every object.

```
interface DXPSocialMediaLinkDataProps extends SocialMediaLinkDataProps {
  donateUrl?: string;
}
```

If it's a type you can do
```
type DXPSocialMediaLinkDataProps = SocialMediaLinkDataType & {
  donateUrl?: string;
}
```

I find it easier to read interfaces.